Src/AutoFixture/Fixture.cs

```diff
@@ -76,54 +76,47 @@ public Fixture(ISpecimenBuilder engine, MultipleRelay multiple)
                                         new ExactTypeSpecification(
                                             typeof(ISpecimenBuilder)))),
+
                                 new StableFiniteSequenceRelay(),
-                                new FilteringSpecimenBuilder(
-                                    new Postprocessor(
-                                        new MethodInvoker(
-                                            new ModestConstructorQuery()),
-                                        new DictionaryFiller()),
-                                    new ExactTypeSpecification(typeof(Dictionary<,>))),
-                                new FilteringSpecimenBuilder(
-                                    new Postprocessor(
-                                        new MethodInvoker(
-                                            new ModestConstructorQuery()),
-                                        new DictionaryFiller()),
-                                    new ExactTypeSpecification(typeof(SortedDictionary<,>))),
-                                new FilteringSpecimenBuilder(
-                                    new Postprocessor(
-                                        new MethodInvoker(
-                                            new ModestConstructorQuery()),
-                                        new DictionaryFiller()),
-                                    new ExactTypeSpecification(typeof(SortedList<,>))),
-                                new FilteringSpecimenBuilder(
-                                    new MethodInvoker(
-                                        new EnumerableFavoringConstructorQuery()),
-                                    new ExactTypeSpecification(typeof(ObservableCollection<>))),
-                                new FilteringSpecimenBuilder(
-                                    new MethodInvoker(
-                                        new ListFavoringConstructorQuery()),
-                                    new ExactTypeSpecification(typeof(Collection<>))),
-                                new FilteringSpecimenBuilder(
-                                    new MethodInvoker(
-                                        new EnumerableFavoringConstructorQuery()),
-                                    new ExactTypeSpecification(typeof(HashSet<>))),
-                                new FilteringSpecimenBuilder(
-                                    new MethodInvoker(
-                                        new EnumerableFavoringConstructorQuery()),
-                                    new ExactTypeSpecification(typeof(SortedSet<>))),
-                                new FilteringSpecimenBuilder(
-                                    new MethodInvoker(
-                                        new EnumerableFavoringConstructorQuery()),
-                                    new ExactTypeSpecification(typeof(List<>))),
+
+                                MakeQueryBasedDictionaryBuilder(
```

Comment:
This is adding a level of indirection though, isn't it?

Reply:
Yep, it's adding. However that is not a big issue, at that heavily improves the readability. Instead parsing the whole graph to understand the supported types, you can simply look at the essential.

Previously we had the same for the "fixed builders":
https://github.com/AutoFixture/AutoFixture/blob/0e2075464a23f7cec5ddec92dfeaff5b5f32cf6e/Src/AutoFixture/Fixture.cs#L124-L126

I've extended that idea more and extracted repeated graph to a sub-method.